addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverter.java

```diff
@@ -86,93 +91,167 @@ public SwaggerConverter(
+        List<String> urls = new ArrayList<>();
+        for (Server server : servers) {
+            String url = server.getUrl();
+            if (server.getVariables() != null) {
+                for (Map.Entry<String, ServerVariable> entry : server.getVariables().entrySet()) {
+                    // default is always set
+                    url = url.replace("{" + entry.getKey() + "}", entry.getValue().getDefault());
+                }
+            }
+            urls.add(url);
+        }
+        return urls;
     }
 
     public List<String> getErrorMessages() {
-        SwaggerDeserializationResult res = new SwaggerParser().readWithInfo(this.defn);
-        if (res != null) {
+        ParseOptions options = new ParseOptions();
+        options.setResolveFully(true);
+        SwaggerParseResult res =
+                new OpenAPIV3Parser().readContents(this.defn, new ArrayList<>(), options);
+        if (res.getOpenAPI() == null) {
+            // try v2
+            res = new OpenAPIParser().readContents(this.defn, new ArrayList<>(), options);
+        }
+        if (res != null && res.getMessages() != null) {
             errors.addAll(res.getMessages());
         }
         errors.addAll(this.generators.getErrorMessages());
         return errors;
     }
 
-    private void addOperations(
-            Swagger swagger, Scheme scheme, String host, List<OperationModel> operations) {
-        switch (scheme) {
-            case HTTP:
-            case HTTPS:
-                for (Map.Entry<String, Path> entry : swagger.getPaths().entrySet()) {
-                    String url =
-                            generators
-                                    .getPathGenerator()
-                                    .getBasicURL(
-                                            scheme, host, swagger.getBasePath(), entry.getKey());
-                    Path path = entry.getValue();
-                    operations.addAll(operationHelper.getAllOperations(path, url));
-                }
-                break;
-            case WS:
-            case WSS:
-                // Dont currently support these
-                break;
+    private void addOperations(OpenAPI openApi, String url, List<OperationModel> operations) {
+        for (Map.Entry<String, PathItem> entry : openApi.getPaths().entrySet()) {
+            PathItem path = entry.getValue();
+            // to avoid having two consecutive slashes
+            if (url.endsWith("/")) url = url.substring(0, url.length() - 1);
+            operations.addAll(operationHelper.getAllOperations(path, url + entry.getKey()));
         }
     }
+
+    private static boolean isRelativeUrl(String url) {
+        // remove curly braces, those are illegal in URI
+        // but can be present in openAPI definition path variables
+        String sanitizedUrl = url.replaceAll("\\{", "");
+        sanitizedUrl = sanitizedUrl.replaceAll("}", "");
+
+        return !URI.create(sanitizedUrl).isAbsolute();
```

Comment:
For consistency this should call the `URI` constructor and wrap the exception in a `SwaggerException` instead of letting it be `IllegalArgumentException`.